cmd/snapd-generator/main.c

```diff
@@ -127,6 +127,52 @@ static bool executable_exists(const char *name) {
 	return false;
 }
 
+static bool is_snap_try_snap_unit(const char *units_dir, const char *mount_unit_name) {
+	char fname[PATH_MAX + 1] = { 0 };
+	sc_must_snprintf(fname, sizeof fname, "%s/%s", units_dir, mount_unit_name);
+	FILE *f SC_CLEANUP(sc_cleanup_file) = NULL;
+	f = fopen(fname, "r");
+	if (!f) {
+		// not really expected
+		fprintf(stderr, "cannot open mount unit %s: %m\n", fname);
+		return false;
+	}
+
+	fseek(f, 0, SEEK_END);
+	long size = ftell(f);
+	fseek(f, 0, SEEK_SET);
+
+	// read entire mount unit file into buffer
+	char *buffer = NULL;
+	sc_cleanup_string(&buffer);
+	buffer = malloc(size + 1);
+	size_t n = fread(buffer, size, 1, f);
+	if (ferror(f) != 0) {
+        fprintf(stderr, "cannot read mount unit %s\n", fname);
+		return false;
+    }
```

Comment:
there's some indentation issues, make sure to run `make fmt`, it will likely produce a lot of changes, but include the relevant chunk in cmd/snapd-generator/main.c only.